lib/disco_log/error.ex

```diff
@@ -13,18 +13,19 @@ defmodule DiscoLog.Error do
           source_function: String.t(),
           context: any(),
           stacktrace: Stacktrace.t(),
-          fingerprint: String.t()
+          fingerprint: String.t(),
+          source_url: String.t() | nil
         }
 
-  defstruct ~w(kind reason source_line source_function context stacktrace fingerprint)a
+  defstruct ~w(kind reason source_line source_function context stacktrace fingerprint source_url)a
 
-  def new(exception, stacktrace, context, client_app) do
+  def new(exception, stacktrace, context, config) do
     {kind, reason} = normalize_exception(exception, stacktrace)
     stacktrace = Stacktrace.new(stacktrace)
-    source = Stacktrace.source(stacktrace, client_app)
+    source = Stacktrace.source(stacktrace, config[:otp_app])
```

Comment:
nit: I'd vote for using a dot notation `config.otp_app` throughout the module to make code more assertive, since we expect validated config everywhere. It's nice to not worry about `nil` values :)